android-client-sdk/src/main/java/com/devcycle/sdk/android/api/DVCClient.kt

```diff
@@ -337,44 +337,47 @@ class DVCClient private constructor(
 
     private suspend fun handleQueuedConfigRequests() {
         configRequestMutex.withLock {
-            if (configRequestQueue.isEmpty()) {
-                return@withLock
-            }
-
-            var latestUserAndCallback: UserAndCallback = configRequestQueue.remove()
-            val callbacks: MutableList<DVCCallback<Map<String, Variable<Any>>>> =
-                mutableListOf()
-
-            if (latestUserAndCallback.callback != null) {
-                callbacks.add(latestUserAndCallback.callback!!)
-            }
-            val itr = configRequestQueue.iterator()
-
-            while (itr.hasNext()) {
-                val userAndCallback = itr.next()
-                if (userAndCallback.now > latestUserAndCallback.now) {
-                    latestUserAndCallback = userAndCallback
+            /*
+             * In case a queued request callback adds another item to the queue
+             * iterate until the queue is empty
+             */
+            while (!configRequestQueue.isEmpty()) {
```

Comment:
kind of confused about the premise of the issue here, because if you call `resetUser` in the callback of the first method like you do in your example, it should get executed immediately and not added to the queue since the lock would be available.

Reply:
The missing piece in that example may be that this is called outside of the `client.onInitialized`, so the identifyUser fetch is added to the queue, and the `handleQueuedConfigRequests` logic doesn't handle items that are added to the queue by a queued callback.

The code below fetches a config and calls all of the callbacks in the queue — then the lock is released. If one of those callbacks queues another request it's left in the queue
```
while (itr.hasNext()) {
    val userAndCallback = itr.next()
    if (userAndCallback.now > latestUserAndCallback.now) {
        latestUserAndCallback = userAndCallback
    }
    if (userAndCallback.callback != null) {
        callbacks.add(userAndCallback.callback)
    }
    itr.remove()
}

val localUser = latestUserAndCallback.user

try {
    fetchConfig(localUser)
    config?.variables?.let { v ->
        callbacks.forEach {
            it.onSuccess(v)
        }
    }
}
```